sway-core/src/asm_generation/abstract_instruction_set.rs

```diff
@@ -64,12 +64,13 @@ impl AbstractInstructionSet {
         //     MOVE t, a        MOVE b, a
         //     MOVE b, t   =>   USE  b
         //     USE  b
-
         loop {
             // Gather all the uses for each register.
-            let uses: BTreeSet<&VirtualRegister> =
-                self.ops.iter().fold(BTreeSet::new(), |mut acc, op| {
-                    acc.append(&mut op.use_registers());
+            let uses: HashSet<&VirtualRegister> =
+                self.ops.iter().fold(HashSet::new(), |mut acc, op| {
+                    for u in &op.use_registers() {
+                        acc.insert(u);
+                    }
```

Comment:
Is the for-loop faster than `append()`?  `append()` is more concise so if there's no perf gain...

Reply:
Wait does append work for `HashMap`? I thought it didn't

Reply:
Oh, I thought it would.  Ugh, sorry.  I thought it just took the slice and inserted items, but it's not there, so obviously not...

Reply:
It _does_ have extend, so maybe `acc.extend(op.use_registers().iter())` might work.  Whatever.